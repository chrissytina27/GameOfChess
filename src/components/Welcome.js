import React from "react";
import PropTypes from "prop-types";
import "../css/welcome.css";

const Welcome = props => {
  const p1Ref = React.createRef();
  const p2Ref = React.createRef();

  const submitNames = e => {
    e.preventDefault();
    if (!(p1Ref.current.value && p2Ref.current.value)) {
      alert("Please enter valid player names");
      return;
    }
    props.playGame(p1Ref.current.value, p2Ref.current.value);
  };

  return (
    <div className="welcome" data-testid="welcome">
      <header className="title">
        <h1>The Game of Chess</h1>
      </header>
      <div className="container">
        <form name="playerNames" onSubmit={submitNames}>
          <label htmlFor="player1">Player 1: </label>
          <input
            type="text"
            id="player1"
            name="player1"
            placeholder="Player 1's Name"
            ref={p1Ref}
          ></input>
          <label htmlFor="player2">Player 2: </label>
          <input
            type="text"
            name="player2"
            id="player2"
            placeholder="Player 2's Name"
            ref={p2Ref}
          ></input>
          <br />
          <button data-testid="startBtn" className="btn" type="submit">
            Start
          </button>
        </form>
      </div>
    </div>
  );
};
Welcome.propTypes = {
  playGame: PropTypes.func
};
export default Welcome;
