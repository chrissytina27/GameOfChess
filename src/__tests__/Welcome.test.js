import Welcome from "../components/Welcome";
import React from "react";
import ReactDOM from "react-dom";
import { cleanup, render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);
it("renders with crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Welcome></Welcome>, div);
});
// it("renders component correctly", () => {
//   const { getByTestId } = render(<Welcome />);
//   const startBtn = getByTestId("startBtn");
//   console.log(startBtn);
//   // expect(getByTestId("startBtn")).toHaveTextContent("");
// });
it("matches snapshot", () => {
  const tree = renderer.create(<Welcome />).toJSON();
  expect(tree).toMatchSnapshot();
});
