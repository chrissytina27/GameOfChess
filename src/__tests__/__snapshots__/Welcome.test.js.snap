// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="welcome"
  data-testid="welcome"
>
  <header
    className="title"
  >
    <h1>
      The Game of Chess
    </h1>
  </header>
  <div
    className="container"
  >
    <form
      name="playerNames"
      onSubmit={[Function]}
    >
      <label
        htmlFor="player1"
      >
        Player 1: 
      </label>
      <input
        id="player1"
        name="player1"
        placeholder="Player 1's Name"
        type="text"
      />
      <label
        htmlFor="player2"
      >
        Player 2: 
      </label>
      <input
        id="player2"
        name="player2"
        placeholder="Player 2's Name"
        type="text"
      />
      <br />
      <button
        className="btn"
        data-testid="startBtn"
        type="submit"
      >
        Start
      </button>
    </form>
  </div>
</div>
`;
